scrapper.py

```diff
@@ -27,71 +40,233 @@ class UnknownConfigError(Exception):
+                    raise BadStatusCode
+            except BadStatusCode:
+                continue
+            else:
+                links = self._extract_url(soup_menu_page)
+                links = [link for link in links if link not in self.urls]
+                if links:
+                    limit = min(len(links), self.max_articles_per_seed)
+                    for link in links[:limit]:
+                        if len(self.urls) < self.total_max_articles:
+                            self.urls.append(link)
+                if len(self.urls) == self.total_max_articles:
+                    break
+        return self.urls
+
+    def get_search_urls(self):
+        """
+        Returns seed_urls param
+        """
+        return self.seed_urls
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler
+    """
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        if os.path.exists(os.path.join(PROJECT_ROOT, 'intermediate_results.json')):
+            with open(os.path.join(PROJECT_ROOT, 'intermediate_results.json'),
+                      'r', encoding='utf-8') as file:
+                results = json.load(file)
+            if results['seed_urls']:
+                self.seed_urls = results['seed_urls']
+            if results['article_urls']:
+                self.urls = results['article_urls']
+        for url in self.get_search_urls():
+            try:
+                response = requests.get(url, headers=headers)
+                if response.status_code == 200:
+                    sleep(random.randrange(3, 6))
+                    soup_menu_page = BeautifulSoup(response.content, 'lxml')
+                else:
+                    raise BadStatusCode
+            except BadStatusCode:
+                continue
+            else:
+                links = self._extract_url(soup_menu_page)
+                links = [link for link in links if link not in self.urls]
+                if links:
+                    limit = min(len(links), self.max_articles_per_seed)
+                    for link in links[:limit]:
+                        if len(self.urls) < self.total_max_articles:
+                            self.urls.append(link)
+                    with open(os.path.join(PROJECT_ROOT, 'intermediate_results.json'),
+                              'r', encoding='utf-8') as file:
+                        results = json.load(file)
+                    results['seed_urls'].remove(url)
+                    results['article_urls'] = self.urls
+                    with open(os.path.join(PROJECT_ROOT, 'intermediate_results.json'), 'w', encoding='utf-8') as file:
+                        json.dump(results, file)
+                if len(self.urls) == self.total_max_articles:
+                    break
+        return self.urls
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        for link in self.seed_urls:
```

Comment:
the method is a spagetti with initialization, parsing and keeping the state. be SOLID (!)